src/main/java/beam/analysis/PathTraversalSpatialTemporalTableGenerator.java

```diff
@@ -333,13 +329,13 @@ public void printDataToFile(String path) {
                         sb.append("\t");
                         sb.append(convertVehicleType(vehicleType));
                         sb.append("\t");
-                        sb.append(energyConsumption[i].get(r5Link.linkId, vehicleAndFuelType));
+                        sb.append(energyConsumption.get(i).get(r5Link.linkId, vehicleAndFuelType));
                         sb.append("\t");
                         sb.append(fuelType);
                         sb.append("\t");
-                        sb.append(Math.round(numberOfVehicles[i].get(r5Link.linkId, vehicleAndFuelType)));
+                        sb.append(Math.round(numberOfVehicles.get(i).get(r5Link.linkId, vehicleAndFuelType)));
                         sb.append("\t");
-                        sb.append(Math.round(numberOfPassengers[i].get(r5Link.linkId, vehicleAndFuelType)));
+                        sb.append(Math.round(numberOfPassengers.get(i).get(r5Link.linkId, vehicleAndFuelType)));
```

Comment:
Two changes:
1-Extract the tab character as a local final variable or a class level constant
2-Consider chaining all appends without repeating `sb`